File: 122.java
// LeetCode 122 Best time to buy and sell stock II
// Reference: https://leetcode.com/problems/best-time-to-buy-and-sell-stock-ii/solutions/4053621/beginner-friendly-solution-beats-100/?envType=featured-list&envId=top-interview-questions?envType=featured-list&envId=top-interview-questions

class Solution {
    public int maxProfit(int[] prices) {
        int profit= 0;
        for (int i = 1; i < prices.length; i++){
            if (prices[i] > prices[i-1]){
                profit += prices[i] - prices[i-1];
            }
        }
        return profit;
    }
}

File: 122.py
# LeetCode 122 Best time to buy and sell stock II
# Reference:https://www.youtube.com/watch?v=3SJ3pUkPQMc

class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        profit = 0

        for i in range(1, len(prices)):
            if prices[i] > prices[i-1]:
                profit += (prices[i] - prices[i-1])
        return profit